lib/fs.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const mkdir = (dir, mode, cb) => {
+  fs.access(dir, fs.constants.F_OK, err => {
+    if (err && err.code === 'ENOENT') {
+      mkdir(path.dirname(dir), mode, err => {
+        if (err) cb(err);
+        else fs.mkdir(dir, cb);
+      });
+    } else {
+      cb(err);
+    }
+  });
+};
+
+const version = process.versions.node.split('.').map(el => parseInt(el));
+const mkdirp = (...args) => {
```

Comment:
I don't like rest syntax usage here. I will look much simpler like this
```js
(path, mode, cb) => {
  if (typeof mode === 'function') {
    cb = mode;
    mode = '0777';
  }
```

Reply:
Also, you reverted back to checking version every time you call the function which is something we try to avoid.